fastlane/Fastfile

```diff
@@ -826,6 +826,34 @@ end
       xctestrun: testPlanPath,
       try_count: options[:try_count],
       output_directory: "build/results",
-      result_bundle: true
+      result_bundle: true,
+      fail_build: false
     )
+
+    # Generate reports with trainer and delete scan ones.
+    generate_trainer_reports(summary[:report_files])
+    delete_scan_reports(summary[:report_files])
+
+    # Manually fail build if test have failed.
+    UI.test_failure!("Tests have failed") if summary[:failed_testcount] > 0
+  end
+
+  # -----------------------------------------------------------------------------------
+  # Geneate trainer reports for each xcresult report file.
+  # -----------------------------------------------------------------------------------
+  def generate_trainer_reports(report_files)
+    report_files.each do | report_path |
+      next if File.extname(report_path) != ".xcresult"
+      trainer(path: report_path, fail_build: false)
+    end
   end
+
+  # -----------------------------------------------------------------------------------
+  # Deletes html & junit reports generated by scan.
+  # -----------------------------------------------------------------------------------
+  def delete_scan_reports(report_files)
+    report_files.each do | report_path |
```

Comment:
💄 nit: the rubocop linter and common guidelines advocate for no spaces around `|` bars:
```suggestion
    report_files.each do |report_path|
```